File: Semestre_2/Metodos_Estadisticos/Entregas/Matlab/T04_Serie_de_Taylor.m
clear;
close all;
clc;
syms x

% --- Función polinómica ---
f_poly = x^4 - 3*x^2 + 2;
a1 = 1;                 % punto de expansión
n1 = 4;                 % orden (>= grado del polinomio para exactitud)
t_poly = taylor(f_poly, x, a1, 'Order', n1+1);
fprintf('Polinómica: f(x) = x^4 - 3x^2 + 2, expandida en a = %g, orden = %d\n', a1, n1);
disp(t_poly);

% Tabla polinómica
x_vals = (-2:0.5:3).';
f_poly_num = matlabFunction(f_poly);
t_poly_num = matlabFunction(t_poly);
tab_poly = table(x_vals, f_poly_num(x_vals), t_poly_num(x_vals), ...
                 abs(f_poly_num(x_vals) - t_poly_num(x_vals)), ...
                 'VariableNames', {'x','f_real','f_taylor','error_abs'});
disp('Tabla polinómica:');
disp(tab_poly);

% --- Función trigonométrica ---
f_trig = sin(x);
a2 = 0;                 % Maclaurin
n2 = 9;                 % orden impar para sin (hasta x^9)
t_trig = taylor(f_trig, x, a2, 'Order', n2+1);
fprintf('\nTrigonométrica: f(x) = sin(x), expandida en a = %g, orden = %d\n', a2, n2);
disp(t_trig);

% Tabla trigonométrica
x_vals2 = (-pi:pi/6:pi).';
f_trig_num = matlabFunction(f_trig);
t_trig_num = matlabFunction(t_trig);
tab_trig = table(x_vals2, f_trig_num(x_vals2), t_trig_num(x_vals2), ...
                 abs(f_trig_num(x_vals2) - t_trig_num(x_vals2)), ...
                 'VariableNames', {'x','f_real','f_taylor','error_abs'});
disp('Tabla trigonométrica:');
disp(tab_trig);

File: Semestre_2/Metodos_Estadisticos/Entregas/Python/T04_Serie_de_Taylor.py
import numpy as np
import pandas as pd
from sympy import symbols, sin, series, lambdify

# Variable simbólica
x = symbols('x')

# --- Función polinómica ---
f_poly = x**4 - 3*x**2 + 2
a1 = 1          # punto de expansión
n1 = 4          # orden

# Serie de Taylor
t_poly = series(f_poly, x, a1, n1+1).removeO()

print(f"Polinómica: f(x) = x^4 - 3x^2 + 2, expandida en a = {a1}, orden = {n1}")
print(t_poly)

# Tabla polinómica
x_vals = np.arange(-2, 3.5, 0.5)

f_poly_num = lambdify(x, f_poly, 'numpy')
t_poly_num = lambdify(x, t_poly, 'numpy')

f_real_vals = f_poly_num(x_vals)
f_taylor_vals = t_poly_num(x_vals)
error_abs = np.abs(f_real_vals - f_taylor_vals)

tab_poly = pd.DataFrame({
    'x': x_vals,
    'f_real': f_real_vals,
    'f_taylor': f_taylor_vals,
    'error_abs': error_abs
})

print("\nTabla polinómica:")
print(tab_poly)

# --- Función trigonométrica ---
f_trig = sin(x)
a2 = 0
n2 = 9

t_trig = series(f_trig, x, a2, n2+1).removeO()

print(f"\nTrigonométrica: f(x) = sin(x), expandida en a = {a2}, orden = {n2}")
print(t_trig)

# Tabla trigonométrica
x_vals2 = np.arange(-np.pi, np.pi + np.pi/6, np.pi/6)

f_trig_num = lambdify(x, f_trig, 'numpy')
t_trig_num = lambdify(x, t_trig, 'numpy')

f_real_vals2 = f_trig_num(x_vals2)
f_taylor_vals2 = t_trig_num(x_vals2)
error_abs2 = np.abs(f_real_vals2 - f_taylor_vals2)

tab_trig = pd.DataFrame({
    'x': x_vals2,
    'f_real': f_real_vals2,
    'f_taylor': f_taylor_vals2,
    'error_abs': error_abs2
})

print("\nTabla trigonométrica:")
print(tab_trig)
